objectbox/lib/flatbuffers/flat_buffers.dart

```diff
@@ -703,18 +703,47 @@ class Builder {
   }
 
   int _writeString(String value) {
-    // TODO(scheglov) optimize for ASCII strings
-    List<int> bytes = utf8.encode(value);
-    int length = bytes.length;
+    // [utf8.encode()] is slow (up to at least Dart SDK 2.13). If the given
+    // string is ASCII we can just write it directly, without any conversion.
+    final originalTail = _tail;
+    if (!_tryWriteASCIIString(value)) {
+      // reset the output buffer position for [_writeUTFString()]
+      _tail = originalTail;
+      _writeUTFString(value);
+    }
+    return _tail;
+  }
+
+  // Try to write the string as ASCII, return false if there's a non-ascii char.
+  @pragma('vm:prefer-inline')
+  bool _tryWriteASCIIString(String value) {
+    _prepare(4, 1, additionalBytes: value.length + 1);
+    final length = value.length;
+    var offset = _buf.lengthInBytes - _tail + 4;
+    for (var i = 0; i < length; i++) {
+      // utf16 code unit, e.g. for '†' it's [0x20 0x20], which is 8224 decimal.
+      final char = value.codeUnitAt(i);
+      if (char >= 128) {
```

Comment:
Maybe add a note that ASCII characters go from 0x00 to 0x7F = 0 to 127 decimal as well?

Reply:
OK, added